src/main/java/com/indix/client/impl/IndixApiClientImpl.java

```diff
@@ -245,6 +244,7 @@ public OffersSearchResult getProductsOffersPremium(Query query) throws IndixApiE
                     });
             return searchIndixApiResponse.getResult();
         } catch (IndixApiException iae) {
+            logger.error("getProductsOffersPremium failed: " + iae.getMessage());
             throw iae;
         } catch (Exception e) {
             logger.error("getProductsOffersPremium failed: " + e.getMessage());
```

Comment:
@shrechak Why do we have 2 catch clause? What's the difference between them? Also, can you explain the purpose of throwing `InternalServerException` for all Exception ?